src/kkos2-display-bundle/ExternalData/EventfeedHelper.php

```diff
@@ -53,32 +53,58 @@ public function setSlide(SlidesInSlide $slide, string $slideType) {
   }
 
   /**
-   * @param string $url Url to feed
-   * @param integer $numItems Number of items to return from the feed
-   * @param array $queryData
+   * Fetch data from feeds.
    *
-   * @return array|mixed
+   * Feeds are provieded via the "datafeed_url" option on the slide, it contains
+   * zero or more feed urls separated by newlines.
+   *
+   * @return array
+   *   An array of json objects.
    */
   public function fetchData() {
-    $url = $this->slide->getOption('datafeed_url', '');
-    if (!$this->validateFeedUrl()) {
-      throw new \Exception("$url is not a valid {$this->slideType} url.");
+    $datafeed_urls = $this->slide->getOption('datafeed_url', '');
+    if (empty($datafeed_urls)) {
+      return [];
     }
-    $query = [];
-    $filterDisplay = $this->slide->getOption('datafeed_display', '');
-    if (!empty($filterDisplay)) {
-      $query = [
-        'display' => $filterDisplay,
-      ];
+
+    // Assume we get multiple feeds.
+    $urls = array_map(
+      "trim",
+      explode("\n", $datafeed_urls)
+    );
+
+    // Go trough reach feed, fetch the contents and merge into a common
+    // response.
+    $feed_data = [];
+    foreach ($urls as $feed_url) {
+      if (!$this->validateFeedUrl($feed_url)) {
+        throw new \Exception("$feed_url is not a valid {$this->slideType} url.");
+      }
```

Comment:
When dealing with multiple items throwing exceptions for one can be tricky as it will halt all processing while other items might be valid. It this case it seems as if the exception will not even reach the administrative user and alert them of the problem as it occurs during data fetching.

The class has a logger injected. Have you considered logging the error instead and continue processing?

Reply:
Good idéa, I'll switch it to logging instead.